node/src/components/consensus/era_supervisor.rs

```diff
@@ -180,45 +183,51 @@ where
 
         let bonded_eras = era_supervisor.bonded_eras();
         let era_ids: Vec<EraId> = era_supervisor
-            .iter_past(current_era, era_supervisor.bonded_eras().saturating_mul(3))
+            .iter_past(current_era, bonded_eras.saturating_mul(3))
             .collect();
 
-        // Asynchronously collect the information needed to initialize all recent eras.
-        let effects = async move {
-            info!(?era_ids, "collecting key blocks and booking blocks");
-
-            let key_blocks = effect_builder
-                .collect_key_block_headers(era_ids.iter().cloned())
-                .await
-                .expect("should have all the key blocks in storage");
-
-            let booking_blocks = collect_booking_block_hashes(
-                effect_builder,
-                era_ids.clone(),
-                auction_delay,
-                activation_era_id,
-            )
-            .await;
-
-            if current_era > activation_era_id.saturating_add(bonded_eras.saturating_mul(2).into())
-            {
-                // All eras can be initialized using the key blocks only.
-                (key_blocks, booking_blocks, Default::default())
-            } else {
-                let activation_era_validators = effect_builder
-                    .get_era_validators(activation_era_id)
-                    .await
-                    .unwrap_or_default();
-                (key_blocks, booking_blocks, activation_era_validators)
+        // Collect the information needed to initialize all recent eras.
+        let mut key_blocks = HashMap::new();
+        let mut booking_blocks = HashMap::new();
+        for era_id in era_ids {
+            let minimum_era = era_supervisor
+                .protocol_config
+                .last_emergency_restart
+                .unwrap_or_else(|| EraId::from(0u64));
+            let key_block_era_id = era_id.saturating_sub(1);
+            if key_block_era_id < minimum_era {
+                continue;
+            }
```

Comment:
This shouldn't even be possible, should it? `era_ids` shouldn't contain an era ID that's less than `minimum_era + 1` because `iter_past()` only starts at `last_emergency_restart + 1` (or 1 if there was no emergency restart).

Reply:
The old async code had the check so I just replicated that...